paddle/math/Matrix.h

```diff
@@ -370,7 +370,7 @@ class Matrix : public BaseMatrix {
    * allocate matTrans' memory outside, then set memAlloc as false;
    * else set as true.
    */
-  virtual void transpose(MatrixPtr matTrans, bool memAlloc) {
+  virtual void transpose(MatrixPtr& matTrans, bool memAlloc) {
```

Comment:
I think they have to be changed.
The original design is not good, obviously in the code does not has `transpose(dst, true)` this usage. Otherwise this problem may be discovered earlier.